packages/compiler-sfc/src/script/resolveType.ts

```diff
@@ -764,6 +766,21 @@ function innerResolveTypeReference(
           !ns.declare,
         )
       }
+    } else {
+      let importType: Import
+      if (
+        (importType = scope.imports[name[0]]) &&
+        importType.imported === '*'
+      ) {
+        const importPath = scope.resolvedImportSources[importType.source]
+        return innerResolveTypeReference(
+          ctx,
+          fileToScope(ctx, normalizePath(importPath)),
+          name.length > 2 ? name.slice(1) : name[name.length - 1],
```

Comment:
There's only a single test case for this, with `name.length` equal to 2. The `name.slice(1)` part isn't currently tested.

This line seems to be repeating some logic from 15 lines earlier. Assuming that isn't coincidental, perhaps it should be refactored to share the logic?

Reply:
Updated~
Please feel free to review.
Thanks.